### PR TITLE
Truncate NFT names after a max number of chars

### LINKED ISSUE #248: Add limit on showing NFT token names
I guess we should limit what we showing on gallery view as people are hijacking names

image

### PR DESCRIPTION
**Pull request by riton**
* Fixes #248
* NFT names longer than the max limit are truncated and a `...` string is appended

### CODE REVIEW

**Comment by yangwao** [comment]
Good one! Love it! 
Maybe we could have it as a global filter as we sense we need to use it elsewhere 
It may fix also issue with 3D models https://github.com/kodadot/nft-gallery/issues/242
but not sure how we could better wrap longer names, idea @riton ?

**Comment by riton** [comment]
I've migrated the function to a _string library_ as requested.

I've also added a `title` on the parent `<p>` element that shows the full NFT name on _mouseover_ :
full_nft_tag_title

**Comment by yangwao** [comment]
Looks almost good but another size/viewport, maybe it could be dynamically truncated? 
Have you tried to play with that around?
Should be doing better when it's gonna be paid :)
I mean that would determine if you will get $50 or $100 for this :)
It's up to you.

image

File: dashboard/src/components/rmrk/Gallery/Gallery.vue [hunk 1]
```diff
@@ -36,12 +36,13 @@
               <div class="card-content">
                 <p
                   v-if="!isLoading"
-                  class="title is-4 has-text-centered">
+                  class="title is-4 has-text-centered"
+                  :title="nft.name">
                   <router-link v-if="nft.count < 2" :to="{ name: 'nftDetail', params: { id: nft.id }}">
-                    {{ nft.name }}
+                    {{ truncateNFTName(nft.name) }}
```
Comment (vikiival) [inline dashboard/src/components/rmrk/Gallery/Gallery.vue:42]: Please make vue filter out it e.g
`nft.name | truncate`

**Review by vikiival — CHANGES_REQUESTED** [review]

File: dashboard/src/components/rmrk/Gallery/Gallery.vue (hunk 1 above)
Comment (riton) [reply dashboard/src/components/rmrk/Gallery/Gallery.vue]: Should be okay now

**Comment by riton** [comment]
This new version will handle dynamic recalculation based on viewport size.

I'm gonna open another pull request for a _full CSS based_ solution that is, IMHO, better for solving this kind of problem.

**Comment by riton** [comment]
#280 has been opened as an alternative solution

**Comment by yangwao** [comment]
What's with this PR? 
Can we use it for later something, ♻️ ?

**Comment by yangwao** [comment]
@riton we can merge it if you fix conflicts :)

**Comment by vikiival** [comment]
@riton can you update your pr ?

**Comment by yangwao** [comment]
Hey, I'm closing this as seems to lack progress and attention from the PR author.

**Pull request closed without merging**